### PR TITLE
Feature/lit 3173 cayenne network bootstrap

### PR DESCRIPTION
**Pull request by joshLong145**
Refactors the special bootstrapping logic added for `cayenne` which will now use the same code paths as `Habanero` and `Manzano` for its url bootstrapping logic. A new check has been on the port range to see if the node's port is between
`8470 - 8479` as these have been determined to be `https` ports.

### CODE REVIEW

File: packages/contracts-sdk/src/lib/contracts-sdk.ts [hunk 1]
```diff
@@ -915,7 +915,11 @@ export class LitContracts {
 
     const networks = activeValidatorStructs.map((item: any) => {
       let proto = 'https://';
-      if (item.port !== 443) {
+      /**
+       * ports in range of 8470 - 8479 are configured for https on cayenne network
+       * we shouold resepct https on these ports as they are using OpenSSL
+       */
+      if (item.port !== 443 || (item.port < 8480 && item.port > 8469)) {
```
Comment (hwrdtm) [inline packages/contracts-sdk/src/lib/contracts-sdk.ts:922]: Think the ge and le operators are the other way around, and need to use || instead. Or you can just ! The whole expression

Comment (joshLong145) [reply packages/contracts-sdk/src/lib/contracts-sdk.ts]: refactored the above to
```js
if (item.port !== 443 && item.port > 8480 && item.port < 8469) {
```

File: packages/contracts-sdk/src/lib/contracts-sdk.ts [hunk 2]
```diff
@@ -915,7 +915,11 @@ export class LitContracts {
 
     const networks = activeValidatorStructs.map((item: any) => {
       let proto = 'https://';
-      if (item.port !== 443) {
+      /**
+       * ports in range of 8470 - 8479 are configured for https on cayenne network
+       * we shouold resepct https on these ports as they are using OpenSSL
```
Comment (hwrdtm) [inline packages/contracts-sdk/src/lib/contracts-sdk.ts:920]:
```suggestion
       * we shouold resepct https on these ports as they are using trusted ZeroSSL certs
```

File: packages/contracts-sdk/src/lib/contracts-sdk.ts [hunk 3]
```diff
@@ -915,7 +915,11 @@ export class LitContracts {
 
     const networks = activeValidatorStructs.map((item: any) => {
       let proto = 'https://';
-      if (item.port !== 443) {
+      /**
+       * ports in range of 8470 - 8479 are configured for https on cayenne network
```
Comment (hwrdtm) [inline packages/contracts-sdk/src/lib/contracts-sdk.ts:919]:
```suggestion
       * ports in range of 8470 - 8479 are configured for https on custom networks (eg. cayenne)
```

**Review by hwrdtm — APPROVED** [review]

**Comment by Ansonhkg** [comment]
Resolved comment & all e2e tests against cayenne passed. lgtm!

File: packages/contracts-sdk/src/lib/contracts-sdk.ts [hunk 4]
```diff
@@ -915,7 +915,11 @@ export class LitContracts {
...
+       * we shouold resepct https on these ports as they are using OpenSSL
+       */
+      if (item.port !== 443 && item.port > 8480 && item.port < 8469) {
```
Comment (hwrdtm) [inline packages/contracts-sdk/src/lib/contracts-sdk.ts:922]: technically should be `> 8479` and `< 8470` but no big deal

**Pull request merged**